src/careamics/lightning/callbacks/prediction_writer_callback/prediction_writer_callback.py

```diff
@@ -0,0 +1,231 @@
+            write_type=write_type,
+            tiled=tiled,
+            write_func=write_func,
+            write_extension=write_extension,
+            write_func_kwargs=write_func_kwargs,
+        )
+        return cls(write_strategy=write_strategy, dirpath=dirpath)
+
+    def _init_dirpath(self, dirpath):
+        """
+        Initialize directory path. Should only be called from `__init__`.
+
+        Parameters
+        ----------
+        dirpath : pathlib.Path
+            See `__init__` description.
+        """
+        dirpath = Path(dirpath)
+        if not dirpath.is_absolute():
+            dirpath = Path.cwd() / dirpath
+            logger.warning(
+                "Prediction output directory is not absolute, absolute path assumed to"
+                f"be '{dirpath}'"
+            )
+        self.dirpath = dirpath
+
+    def setup(self, trainer: Trainer, pl_module: LightningModule, stage: str) -> None:
+        """
+        Create the prediction output directory when predict begins.
+
+        Called when fit, validate, test, predict, or tune begins.
+
+        Parameters
+        ----------
+        trainer : Trainer
+            PyTorch Lightning trainer.
+        pl_module : LightningModule
+            PyTorch Lightning module.
+        stage : str
+            Stage of training e.g. 'predict', 'fit', 'validate'.
+        """
+        super().setup(trainer, pl_module, stage)
+        if stage == "predict":
+            # make prediction output directory
+            if not self.dirpath.is_dir():
+                logger.info("Making prediction output directory.")
+                self.dirpath.mkdir()
+
+    def write_on_batch_end(
+        self,
+        trainer: Trainer,
+        pl_module: LightningModule,
+        prediction: Any,  # TODO: change to expected type
+        batch_indices: Optional[Sequence[int]],
+        batch: Any,  # TODO: change to expected type
+        batch_idx: int,
+        dataloader_idx: int,
+    ) -> None:
+        """
+        Write predictions at the end of a batch.
+
+        The method of prediction is determined by the attribute `write_strategy`.
+
+        Parameters
+        ----------
+        trainer : Trainer
+            PyTorch Lightning trainer.
+        pl_module : LightningModule
+            PyTorch Lightning module.
+        prediction : Any
+            Prediction outputs of `batch`.
+        batch_indices : sequence of Any, optional
+            Batch indices.
+        batch : Any
+            Input batch.
+        batch_idx : int
+            Batch index.
+        dataloader_idx : int
+            Dataloader index.
+        """
+        # if writing prediction is turned off
+        if not self.writing_predictions:
+            return
+
+        dls: Union[DataLoader, list[DataLoader]] = trainer.predict_dataloaders
+        dl: DataLoader = dls[dataloader_idx] if isinstance(dls, list) else dls
+        ds: Union[IterablePredDataset, IterableTiledPredDataset] = dl.dataset
```

Comment:
Here I would be in favor of using longer variable name (e.g. `dataloader_list`, `dataloader` and `dataset`) for clarity.

Reply:
So I defined `dls` and `dl` mostly for type hinting, otherwise I would have indexed `trainer.predict_dataloaders` directly; I tend to go for short variable names to indicate that they are temporary or not important, but I suppose I can update the variable names for extra clarity. `ds` is actually used later on so it really should be more verbose.